### DISCUSSION TITLE
Conditionally show/hide table column based on Select filters value

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#12609] [Help]
### Package

Table builder

### Package Version

v3.2.71

### How can we help you?

I have two table columns

```
TextColumn::make('monthly_price'),

TextColumn::make('annually_price'),
```

I want to show/hide the columns above based on selected filters value

```
SelectFilter::make('periodicity')
    ->options([
        'monthly' => 'Monthly',
        'yearly' => 'Yearly',
    ])
```

If the monthly is selected hide the `TextColumn::make('annually_price'),`
If the annually is selected hide the `TextColumn::make('monthly_price'),`

**Comment by leandrocfe** [comment]
I think you can inject the ListPage and use `$livewire->tableFilters`

```php
->visible(function (Pages\ListXX $livewire) {
    //$livewire->tableFilters
})
```

**Reply by devhoussam1998** [reply]
could you provide more information please?

**Reply by leandrocfe** [reply]
I think you can get the periodicity value selected with `$livewire->tableFilters`. So you can verify and return true/false.

**Reply by devhoussam1998** [reply]
is there any code example?
